sympy/simplify/fu.py

```diff
@@ -1645,6 +1646,11 @@ def fu(rv, measure=lambda x: (L(x), x.count_ops())):
     rv = sympify(rv)
     if not isinstance(rv, Expr):
         return rv.func(*[fu(a, measure=measure) for a in rv.args])
+    else:
+        n, d = fraction(rv)
+        if d.is_zero is None:
+            if simplify(d) is S.Zero:
+                return nan
```

Comment:
I used it to simplify `sin(x)**2 + cos(x)**2 - 1` to 0 , I guess there must be some other method to do that without using `simplify` which I'm not aware of.

Reply:
`TR5` (or `TR6`) will detect this particular case.

An intermediate approach may be to call `futrig(d)` here which may cover a few more cases. But hard to say if it is worth it in general. If the aim is to avoid cancelling 0/0 it probably makes sense rather than trying to add one special case at a time.